src/Data/LLVM/BitCode/BitString.hs

```diff
@@ -40,27 +44,48 @@ pattern Bytes' n = NumBytes n
 bitCount :: NumBits -> Int
 bitCount (NumBits n) = n
 
+bitCount# :: NumBits -> Int#
+bitCount# (NumBits (I# n#)) = n#
+
 {-# INLINE addBitCounts #-}
 addBitCounts :: NumBits -> NumBits -> NumBits
-addBitCounts (NumBits a) (NumBits b) = NumBits $ a + b
+addBitCounts (NumBits (I# a#)) (NumBits (I# b#)) = NumBits (I# (a# +# b#))
 
 {-# INLINE subtractBitCounts #-}
 subtractBitCounts :: NumBits -> NumBits -> NumBits
-subtractBitCounts (NumBits a) (NumBits b) = NumBits $ a - b
+subtractBitCounts (NumBits (I# a#)) (NumBits (I# b#)) = NumBits (I# (a# -# b#))
 
 {-# INLINE bytesToBits #-}
 bitsToBytes :: NumBits -> (NumBytes, NumBits)
-bitsToBytes (NumBits n) = (NumBytes $ n `shiftR` 3, NumBits $ n .&. 7)
+bitsToBytes (NumBits (I# n#)) = ( NumBytes (I# (n# `uncheckedIShiftRL#` 3#))
+                                , NumBits (I# (n# `andI#` 7#))
+                                )
 
 {-# INLINE bitsToBytes #-}
 bytesToBits :: NumBytes -> NumBits
-bytesToBits (NumBytes n) = NumBits $ n `shiftL` 3
+bytesToBits (NumBytes (I# n#)) = NumBits (I# (n# `uncheckedIShiftL#` 3#))
 
 ----------------------------------------------------------------------
 
 data BitString = BitString
   { bsLength :: !NumBits
-  , bsData   :: !Integer
+  , bsData   :: !Int
+    -- Note: the bsData was originally an Integer, which allows an essentially
+    -- unlimited size value.  However, this adds some overhead to various
+    -- computations, and since LLVM Bitcode is unlikely to ever represent values
+    -- > native size (64 bits) as discrete values.  By changing this to @Int@,
```

Comment:
Is the use of `>` in `> native size [...]` here (which Haddock interprets as inline code) a mistake?

Reply:
It was an auto-formatting in the editor than ended up with that as the first character... good catch!